platform/features/table/src/TableConfiguration.js

```diff
@@ -144,16 +145,32 @@ define(
                 {}).columns || {};
         };
 
+        function equal(obj1, obj2) {
+            var obj1Keys = Object.keys(obj1),
+                obj2Keys = Object.keys(obj2);
+            return (obj1Keys.length === obj2Keys.length) &&
+                    obj1Keys.every(function(key){
+                        //To do a deep equals, could recurse here if typeof
```

Comment:
Comment is not very useful nor accurate; there's really more changes that would be necessary to do a deep equality check than just this; it also distracts from the purpose of this function.  I would remove the comment.

Part of me says to rename the function to more clearly indicate the purpose (detecting if there is a change between two columnConfigurations).  Your call.